tests/core/test_accounts.py

```diff
@@ -409,54 +427,81 @@ def test_eth_account_recover_transaction_from_eth_test(acct, transaction):
 
 
 @pytest.mark.parametrize(
-    'private_key, password, expected_decrypted_key',
+    'private_key, password, kdf, expected_decrypted_key, expected_kdf',
     [
         (
             web3js_key(),
             web3js_password(),
+            None,
             to_bytes(hexstr=web3js_key()),
+            'scrypt'
+        ),
+        (
+            web3js_key(),
+            web3js_password(),
+            'pbkdf2',
+            to_bytes(hexstr=web3js_key()),
+            'pbkdf2'
         ),
         (
             web3js_private_key(web3js_key()),
             web3js_password(),
+            None,
             web3js_private_key(web3js_key()).to_bytes(),
+            'scrypt'
         ),
     ],
-    ids=['hex_str', 'eth_keys.datatypes.PrivateKey']
+    ids=['hex_str', 'hex_str_provided_kdf', 'eth_keys.datatypes.PrivateKey']
 )
-def test_eth_account_encrypt(acct, private_key, password, expected_decrypted_key):
-    encrypted = acct.encrypt(private_key, password)
+def test_eth_account_encrypt(acct, private_key, password, kdf, expected_decrypted_key, expected_kdf):  # noqa: E501
+    if kdf is None:
+        encrypted = acct.encrypt(private_key, password)
+    else:
+        encrypted = acct.encrypt(private_key, password, kdf)
 
     assert encrypted['address'] == '2c7536e3605d9c16a7a3d7b1898e529396a65c23'
     assert encrypted['version'] == 3
+    assert encrypted['crypto']['kdf'] == expected_kdf
 
     decrypted_key = acct.decrypt(encrypted, password)
 
     assert decrypted_key == expected_decrypted_key
 
 
 @pytest.mark.parametrize(
-    'private_key, password, expected_decrypted_key',
+    'private_key, password, kdf, expected_decrypted_key, expected_kdf',
     [
         (
             web3js_key(),
             web3js_password(),
+            None,
+            HexBytes(to_bytes(hexstr=web3js_key())),
+            'scrypt'
+        ),
+        (
+            web3js_private_key(web3js_key()),
+            web3js_password(),
+            'pbkdf2',
             HexBytes(to_bytes(hexstr=web3js_key())),
+            'pbkdf2'
         ),
         (
             web3js_private_key(web3js_key()),
             web3js_password(),
+            None,
             web3js_private_key(web3js_key()).to_bytes(),
+            'scrypt'
         ),
     ],
-    ids=['hex_str', 'eth_keys.datatypes.PrivateKey']
+    ids=['hex_str', 'hex_str_provided_kdf', 'eth_keys.datatypes.PrivateKey']
 )
-def test_eth_account_prepared_encrypt(acct, private_key, password, expected_decrypted_key):
+def test_eth_account_prepared_encrypt(acct, private_key, password, kdf, expected_decrypted_key, expected_kdf):  # noqa: E501
     account = acct.privateKeyToAccount(private_key)
-    encrypted = account.encrypt(password)
+    encrypted = account.encrypt(password) if kdf is None else account.encrypt(password, kdf)
```

Comment:
For readability, I usually recommend something like the following instead of the inline ternary operator:
```py
if kdf is None:
  encrypted = account.encrypt(password)
else:
  encrypted = account.encrypt(password, kdf)
```

The closest thing I can find in the style guide is https://github.com/pipermerriam/ethereum-dev-tactical-manual/blob/master/style-guide.md#one-thing-per-line ... it's probably worth checking if there is broad consensus on that and adding it to the style guide.

An exception might be something like: `num_threads = 0 if cpu_count is None else cpu_count` -- even then I might slightly prefer the other syntax, but not say anything on a PR.

Reply:
Yeah, I thought the ternary operator would be preferred, but for the test above this one ([see here](https://github.com/ethereum/eth-account/pull/38/files/d8ee9af6fd0467d2e25ea028c2ab6e2145d2dcfe#diff-1de692d823115945c2232e1aa3066811R457)) it caused me to break the line length limit so I used the regular conditional blocks. I agree that'd probably be something good to add to the style guide. 👍  I'll switch it to be consistent since the above would need to use the regular conditional blocks unless we ignored flake warnings (which I'm guessing we'd rather not do if it can be avoided). Thanks!

Reply:
Fixed: https://github.com/ethereum/eth-account/pull/38/commits/8aec325d1b02c045143b0106e6a807524c04c6b6#diff-1de692d823115945c2232e1aa3066811R509